freva-data-portal-worker/src/data_portal_worker/load_data.py

```diff
@@ -185,16 +176,24 @@ def from_object_path(
         ).dict()
         expires_in = str_to_int(os.environ.get("API_CACHE_EXP"), 3600)
         status_dict["status"] = 3
+        dset: Optional[xr.Dataset] = None
         self.cache.setex(path_id, expires_in, cloudpickle.dumps(status_dict))
         data_logger.debug("Reading %s", input_path)
         try:
             data_logger.info(input_path)
             dset = load_data(input_path)
             metadata = create_zmetadata(dset)
             status_dict["json_meta"] = jsonify_zmetadata(dset, metadata)
-            status_dict["obj"] = dset
             status_dict["meta"] = metadata
             status_dict["status"] = 0
+            # We need to add the xarray to an extra cache entry because the
+            # status_dict will be loaded by the rest-api, if the xarray dataset
+            # is present the rest-api code will attempt to instanciate the
+            # pickled dataset object and that might fail because we might or
+            # might not have xarray and all the backends instanciated.
+            # Since the xarray dataset object isn't needed anyway byt the
+            # rest-api we simply add it to a cache entry of its own.
```

Comment:
as status dict is changing here, what do u think to update srtting the metadat var in cache before `{path_id}-dset`? i mean adding this line `self.cache.setex(f"{path_id}-metadata", expires_in, cloudpickle.dumps(status_dict))` before dumping the dset

Reply:
No, we shouldn't do that here. Because if something goes wrong we need to update the status dict that something goes wrong.


On the other hand I cannot add the xarray dataset after the try execpt block because I cannot add None  to the cache. I could pickle the None and then add it. But then I have to get the dataset unpickle it and check if it's None. To me its just more straight forward to only add it if its good and then simply check for it's pure existence in the cache.